jquery/src/js/app.js

```diff
@@ -1,43 +1,54 @@
 var TasksBoard = (function ($) {
-    var TasksBoard = {};
-    var self = TasksBoard;
-
-    var $taskInput, $tasksList;
-
-    self.init = function (options) {
-        $taskInput = $(options.inputSelector);
-        $tasksList = $(options.listSelector);
-
-        $(options.inputSelector).keypress(self.addTaskHandler);
-        $(options.listSelector).on('click', '[data-action="remove-task"]', self.removeTaskHandler)
-    }
-
-    self.addTaskHandler = function (e) {
-        if (e.keyCode == 13 && _validateInputText()) {
-            var text = $taskInput.val();
-            var taskTemplate = '<li class="list-group-item"><span data-action="remove-task" class="badge">X</span>' + text + '</li>';
-            $tasksList.append(taskTemplate);
-            $taskInput.val('');
-        }
-    }
-
-    self.removeTaskHandler = function (e) {
-        e.currentTarget.parentElement.remove();
-    }
-
-    function _validateInputText() {
-        $taskInput.toggleClass('input-error', false);
-        $taskInput.attr('title', '');
-        if ($taskInput.val().length == 0) {
-            $taskInput.toggleClass('input-error', true);
-            $taskInput.attr('title', '* Please Fill Out This Field.');
-            $taskInput.tooltip('show');
-            return false
+    TasksBoard = function (options) {
+        this.controls = {
+            $rootEl: $(),
+            $taskInput: $(),
+            $tasksList: $()
+        };
+
+        this.options = options;
+        this._init();
+    };
+
+    TasksBoard.prototype = {
+        _init: function () {
+            var self = this;
+
+            self.controls.$rootEl = $(self.options.rootSelector);
+            self.controls.$taskInput = self.controls.$rootEl.find(self.options.inputSelector);
+            self.controls.$tasksList = self.controls.$rootEl.find(self.options.listSelector);
+
+            self.controls.$taskInput.keypress(self._addTaskHandler.bind(self));
+            self.controls.$tasksList.on('click', '[data-action="remove-task"]', self._removeTaskHandler.bind(self));
```

Comment:
Constant detected ;) Use this pattern for constants:
```javascript
var _MY_CONST = '123';
```
Constants should be at the beginning of a module.